### PR TITLE
[W6.1d][T09-B3] Niu Yunpeng

### PR DESCRIPTION
**Pull request by yunpengn**
- Learning Outcome W6.1d Polymorphism
- Add a polymorphic `isMutating` abstract method to the `Command` superclass and implement it in every its subclass.
- Only save the changes to local storage file when the current command is **mutating**.

**Ready-for-review**

Yunpeng

### CODE REVIEW

File: src/seedu/addressbook/commands/Command.java [hunk 1]
```diff
@@ -40,13 +40,21 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
      * Executes the command and returns the result.
      */
     public CommandResult execute(){
-        throw new UnsupportedOperationException("This method should be implement in child classes");
+        throw new UnsupportedOperationException("This method should be implemented in child classes.");
```
Comment (ngsikai) [inline src/seedu/addressbook/commands/Command.java]: Could this be potentially changed to become an abstract method as well?

File: test/java/seedu/addressbook/logic/LogicTest.java [hunk 2]
```diff
@@ -267,6 +265,10 @@ public void execute_tryToViewMissingPerson_errorMessage() throws Exception {
         expectedAB.addPerson(p2);
 
         addressBook.addPerson(p2);
+        // This statement is necessary because view command does not trigger saving changes after executed.
```
Comment (ngsikai) [inline test/java/seedu/addressbook/logic/LogicTest.java:268]: Consider abstracting this away in a generator/constructor made specially for commands that do not trigger saves.

File: src/seedu/addressbook/commands/Command.java (hunk 1 above)
Comment (ngsikai) [inline src/seedu/addressbook/commands/Command.java]: Could this command also become an abstract method?

**Review by ngsikai — COMMENTED** [review]
Hey Yunpeng, once again a terrific job in covering all ends in the PR. 

I have left some suggestions on how to make the code more user friendly, particularly on how the explicit .saveFile can be abstracted away in tests written for !isMutating commands. This could help devs that are potentially working on more commands that are !isMutating.

Go ahead and close the PR after reading the comments, and see you in class.

File: src/seedu/addressbook/commands/Command.java (hunk 1 above)
Comment (yunpengn) [reply src/seedu/addressbook/commands/Command.java]: Yes, that's a good idea.

Comment (yunpengn) [reply src/seedu/addressbook/commands/Command.java]: The same as above.

**Comment by yunpengn** [comment]
close this PR because the module has moved toward addressbook-level4.

**Pull request closed without merging**